### ISSUE TITLE
Allow binding to vscode commands

### ISSUE THREAD

**Issue by realh** [VSCodeVim/Vim#1031]
------

Please *thumbs-up* 👍 this issue if it personally affects you! You can do this by clicking on the emoji-face on the top right of this post. Issues with more thumbs-up will be prioritized.

-----
Only being able to rebind key sequences to other key sequences with "after" is too limiting. For example, I wanted to implement the missing ZZ command. I tried using "after": {":", "w", "q", "<Ret>" }, but although it opened the command buffer, it didn't insert and execute the wq (disappointing, but understandable). It would be useful if the *KeyBindings* settings had a field for running VSCode commands by name so I could get the desired behaviour with something like "commands": {"workbench.action.files.save", "workbench.action.closeActiveEditor"}.

**Comment by trival** [comment]
It would be wonderful if this allowed the binding `{ "before": [ ":" ], "after": [ "workbench.action.showCommands" ] }` since i never use the Ex commands with vscodeVim and would much prefer the Command Palette with the `:` key for convenience.

**Comment by rebornix** [comment]
@realh @trival is there anything that Code's builtin `keybindings.json` doesn't help as you are mapping key combinations to Code's command, which has nothing to do with this extension.

**Comment by realh** [comment]
I don't know how I would be able to map ZZ to its usual function in vim, which is to save the active tab's content then close the tab. The problems are: (a) It should only do this in vim-normal mode, not in insert mode. (b) It's two separate keystrokes. (c) I think it needs to call two separate Code commands.

But, in general, I think allowing Code commands to be used in vim.otherModesKeyBindingsNonRecursive would make it much more flexible and easier to use. At the moment it only seems possible to remap commands that VSCodeVim already has key-bindings for, and I think some are missing, eg there's no way to save when vim.useCtrlKeys is enabled. And if I disable vim.useCtrlKeys I can't use block selection.

**Comment by realh** [comment]
> And if I disable vim.useCtrlKeys I can't use block selection.

I fixed this by copying the default key binding and deleting the `&& vim.useCtrlKeys` part.

But I still can't get ZZ to do what I want. I tried `{ "before": ["Z", "Z"], "after": ["<C-S>", "<C-W>"] }` but it didn't work. I also tried a lower case s and w. Is this a limitation that the plugin can't currently create new multi-key bindings in "before"?

**Comment by johnfn** [comment]
This is now in master and will be released in 0.4.7. For reference, I was able to implement "Z", "Z" like so (~~though this is kind of a stupid example since I could have just implemented ZZ but w/e~~)

```
    "vim.otherModesKeyBindingsNonRecursive": [
        {
            "before": ["Z", "Z"],
            "commands": [
                {
                    "command": "workbench.action.files.save",
                    "args": []
                },
                {
                    "command": "workbench.action.closeActiveEditor",
                    "args": []
                }
            ]
        }
    ]
        ```
